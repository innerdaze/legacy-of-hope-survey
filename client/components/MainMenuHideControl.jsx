import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Menu from 'grommet/components/Menu'
import Button from 'grommet/components/Button'
import CloseIcon from 'grommet/components/icons/base/Close'

class MainMenuHideControl extends Component {
  constructor(props) {
    super(props)

    this.hideMenu = this.hideMenu.bind(this)
  }

  hideMenu() {
    this.props.hideMenu()
  }

  render() {
    return (
      <Menu direction='row'
        responsive={false}
        inline={true}
        alignSelf='end'>
        <Button plain={true}
          icon={<CloseIcon/>}
          onClick={this.hideMenu} />
      </Menu>
    )
  }
}

MainMenuHideControl.propTypes = {
  hideMenu: PropTypes.func.isRequired
}

MainMenuHideControl.defaultProps = {
  hideMenu: Function.prototype
}

export default MainMenuHideControl
