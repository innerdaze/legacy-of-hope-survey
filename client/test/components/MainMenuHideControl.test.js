import React from 'react'
import MainMenuHideControl from '../../components/MainMenuHideControl'

let wrapper, hideMenu

describe('Testing on MainMenuHideControl...', () => {
  beforeEach(() => {
    hideMenu = jest.fn()
    wrapper = shallow(<MainMenuHideControl hideMenu={hideMenu} />)
  })

  describe('MainMenuHideControl layout', () => {
    test('Expect MainMenuHideControl to have a parent Menu comp and child Button comp', () => {
      expect(wrapper.find('Menu')).toHaveLength(1)
      expect(wrapper.find('Menu').find('Button')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect hideMenu prop function is called if Button is clicked', () => {
      wrapper.find('Button').simulate('click')

      expect(hideMenu).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
