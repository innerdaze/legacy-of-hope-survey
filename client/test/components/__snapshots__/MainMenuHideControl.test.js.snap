// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on MainMenuHideControl... MainMenuHideControl layout Expect MainMenuHideControl to have a parent Menu comp and child Button comp 1`] = `
<Menu
  alignSelf="end"
  closeOnClick={true}
  direction="row"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  inline={true}
  pad="none"
  responsive={false}
>
  <Button
    icon={
      <Close
        responsive={true}
      />
    }
    method="push"
    onClick={[Function]}
    plain={true}
    type="button"
  />
</Menu>
`;

exports[`Testing on MainMenuHideControl... Test on prop Expect hideMenu prop function is called if Button is clicked 1`] = `
<Menu
  alignSelf="end"
  closeOnClick={true}
  direction="row"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  inline={true}
  pad="none"
  responsive={false}
>
  <Button
    icon={
      <Close
        responsive={true}
      />
    }
    method="push"
    onClick={[Function]}
    plain={true}
    type="button"
  />
</Menu>
`;
